src/useNavigationBuilder.tsx

```diff
@@ -92,10 +99,22 @@ export default function useNavigationBuilder(
     [getCurrentState, router.getRehydratedState, router.getInitialState]
   );
 
+  const { childrenOnActionListeners } = React.useMemo<{
```

Comment:
We need ti use `React.useRef` for mutable values like this. `useMemo` isn't guaranteed to persist every render.